### PR TITLE
Reduce number of packges

### PR DESCRIPTION
**Pull request by domasx2**
Rolls most of packages into `@grafana/agent-web`, except for tracing, which gets it's own `@grafana/agent-web-tracing` due otel being so bulky.

Reasons:

* have fewer npm packages in grafana org
* more abstract/generic packages give us more flexibility 
* easier to make sense of, and determine what relates to web and what to node (in the future)

### CODE REVIEW

File: packages/integration-angular/package.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
-  "name": "@grafana/javascript-agent-integration-angular",
+  "name": "@grafana/agent-integration-angular",
   "version": "0.0.0",
   "description": "Grafana JavaScript Agent package that enables easier integration in projects built with Angular.",
   "license": "Apache-2.0",
   "author": "Grafana Labs",
+  "private": true,
```
Comment (bfmatei) [inline packages/integration-angular/package.json:7]: Is it `private` actually?

File: README.md [hunk 2]
```diff
@@ -11,41 +11,24 @@ which can be ran by following the [README.md file](https://github.com/grafana/gr
...
+[@grafana/agent-web](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/web) provides instrumentations, metas and transports for use in web applications. See The [README.md](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/web/README.md) for more information.
 
-1. [@grafana/javascript-agent-meta-browser](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/meta-browser)
-   is a browser package for collecting details about the browser, operating system and device type.
-1. [@grafana/javascript-agent-meta-page](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/meta-page)
-   is a browser package for collecting details about the current page (i.e. the URL, page title etc.).
+### Web-tracing
 
-### Transports
+[@grafana/agent-web-tracing](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/web-tracing) provides implementation for tracing web applications. See The [README.md](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/web-tracing/README.md) for more information.
```
Comment (bfmatei) [inline README.md:24]: 1. I would consider this part of the `Web` section.
2. `web-tracing` vs `tracing-web`. I would advocate for the latter.

File: lerna.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "packages": ["packages/*", "demo"],
-  "version": "0.0.1",
+  "version": "0.0.0",
```
Comment (bfmatei) [inline lerna.json:3]:
```suggestion
  "version": "0.0.0-dev",
```

File: packages/core/src/metas/initialize.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export function initializeMetas(config: Config): Metas {
   };
 
   add('sdk', () => ({
-    name: '@grafana/javascript-agent-core',
+    name: '@grafana/agent-core',
     version: '0.0.1', // TODO: set correct version here
```
Comment (bfmatei) [inline packages/core/src/metas/initialize.ts:20]: This should use the `VERSION` mechanism?

File: packages/integration-react/package.json [hunk 5]
```diff
@@ -1,9 +1,10 @@
 {
-  "name": "@grafana/javascript-agent-integration-react",
+  "name": "@grafana/agent-integration-react",
   "version": "0.0.0",
   "description": "Grafana JavaScript Agent package that enables easier integration in projects built with React.",
   "license": "Apache-2.0",
   "author": "Grafana Labs",
+  "private": true,
```
Comment (bfmatei) [inline packages/integration-react/package.json:7]: Is this `private`?

File: packages/web-tracing/README.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# @grafana/agent-web-tracing
+
+Tracing implementation for web application. @TODO
```
Comment (bfmatei) [inline packages/web-tracing/README.md:3]: TODO

File: packages/web-tracing/package.json [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
   "license": "Apache-2.0",
   "author": "Grafana Labs",
+  "private": true,
   "homepage": "https://github.com/grafana/grafana-javascript-agent",
+  "sideEffects": false,
   "repository": {
     "type": "git",
     "url": "https://github.com/grafana/grafana-javascript-agent.git",
-    "directory": "packages/transport-fetch"
+    "directory": "packages/web"
```
Comment (bfmatei) [inline packages/web-tracing/package.json:13]:
```suggestion
    "directory": "packages/web-tracing"
```

File: packages/web-tracing/package.json [hunk 8]
```diff
@@ -1,14 +1,16 @@
 {
-  "name": "@grafana/javascript-agent-transport-fetch",
+  "name": "@grafana/agent-web-tracing",
   "version": "0.0.0",
-  "description": "Grafana JavaScript Agent browser package for sending the collected data to a specified URL using fetch.",
+  "description": "Grafana JavaScript Agent web tracing implementation.",
   "license": "Apache-2.0",
   "author": "Grafana Labs",
+  "private": true,
```
Comment (bfmatei) [inline packages/web-tracing/package.json:7]: Is this `private`?

**Review by kpelelis — DISMISSED** [review]

**Review by bfmatei — COMMENTED** [review]
nitpicks, suggestions, comments

File: packages/integration-angular/package.json (hunk 1 above)
Comment (domasx2) [reply packages/integration-angular/package.json:7]: would rather not release to npm until it has an implementation, so marked as private

File: README.md (hunk 2 above)
Comment (domasx2) [reply README.md:24]: - [ ] rename to `tracing-web`

File: packages/core/src/metas/initialize.ts (hunk 4 above)
Comment (domasx2) [reply packages/core/src/metas/initialize.ts:20]: - [ ] yes it should

File: packages/web-tracing/README.md (hunk 6 above)
Comment (domasx2) [reply packages/web-tracing/README.md:3]: to be solved in another PR :D

File: packages/core/src/metas/initialize.ts (hunk 4 above)
Comment (domasx2) [reply packages/core/src/metas/initialize.ts:20]: this is covered by meta refac PR, I'll rebase when it's merged

File: lerna.json (hunk 3 above)
Comment (domasx2) [reply lerna.json:3]: let's update version with separate PR, probably when adding release-to-npm mechanism

**Review by bfmatei — APPROVED** [review]

**Pull request merged**